python/ray/serve/api.py

```diff
@@ -536,15 +541,39 @@ def start(detached: bool = False,
 
     Args:
         detached (bool): Whether not the instance should be detached from this
-            script.
-        http_host (str): Host for HTTP servers to listen on. Defaults to
-            "127.0.0.1". To expose Serve publicly, you probably want to set
-            this to "0.0.0.0". One HTTP server will be started on each node in
-            the Ray cluster. To not start HTTP servers, set this to None.
-        http_port (int): Port for HTTP server. Defaults to 8000.
-        http_middlewares (list): A list of Starlette middlewares that will be
-            applied to the HTTP servers in the cluster.
+          script.
+        http_host (str): Deprecated, use http_options instead.
+        http_port (int): Deprecated, use http_options instead.
+        http_middlewares (list): Deprecated, use http_options instead.
+        http_options (Optional[Dict, serve.HTTPOptions]): Configuration options
+          for HTTP proxy. You can pass in a dictionary or HTTPOptions object
+          with fields:
+            - host(str, None): Host for HTTP servers to listen on. Defaults to
+              "127.0.0.1". To expose Serve publicly, you probably want to set
+              this to "0.0.0.0".
+            - port(int): Port for HTTP server. Defaults to 8000.
+            - middlewares(list): A list of Starlette middlewares that will be
+              applied to the HTTP servers in the cluster.
+            - mode(str, serve.config.DeploymentMode): The deployment mode.
+              - "HeadOnly": start on HTTP server on the head node. Serve
+                   assumes the head node is the node you executed serve.start
+                   on. This is the default.
+              - "NodePort": start one HTTP server per node.
```

Comment:
Not exactly sure what the name `NodePort` means. Maybe `EveryNode` or `OnePerNode`?